### PR TITLE
[Tests-only] Adjust public link tests for download, upload and other operations

### PR DESCRIPTION
**Pull request by PVince81**
Enabled some tests and tagged others that are not working yet.

For https://github.com/owncloud/ocis-reva/issues/49.
A lot blocked by https://github.com/owncloud/ocis-reva/issues/282

- ~~REQUIRES: https://github.com/cs3org/reva/pull/822 for some tests to work.~~
- ~~REQUIRES: https://github.com/cs3org/reva/pull/872 for some tests to work.~~
- REQUIRES: https://github.com/cs3org/reva/pull/877

~~Still need to investigate some of the tests tagged with "issue-ocis-reva-49" where the download returns an empty string.~~

### Raised and tagged issues:

- Webdav PUT response must always be 201 for upload-only shares: https://github.com/owncloud/ocis-reva/issues/286
- public link permission handling: https://github.com/owncloud/ocis-reva/issues/292 
- ~~rename and delete not implemented in public links: https://github.com/owncloud/ocis-reva/issues/293~~ resolved in the PR
- prevent sharing root folder: https://github.com/owncloud/ocis-reva/issues/283
- split old and new endpoint tests: https://github.com/owncloud/ocis-reva/issues/282
- OCS API expiration date should allow dates without time component: https://github.com/owncloud/ocis-reva/issues/288
- [x] OCS API support for legacy "publicUpload" attribute https://github.com/owncloud/ocis-reva/issues/289 => fixed in ocis-reva 0.6.0 but tests remains disabled because of other issues
- 500 error instead of 404 when accessing non-existing link: https://github.com/owncloud/ocis-reva/issues/290
- share_with field must contain special value for public links that have a password: https://github.com/owncloud/ocis-reva/issues/294
- single file shares: https://github.com/owncloud/ocis-reva/issues/278

### TODOs:
- [x] investigate why remaining tests tagged with `@issue-ocis-reva-49` are failing with empty downloads in apiSharePublicLink1/createPublicLinkShare.feature
- [x] process the other test files:

### CODE REVIEW

**Comment by PVince81** [comment]
Squashed and pushed. Added some more tickets and excluded tests.

Still, I managed to enable a bunch of tests for public link upload, download, creation, etc. Updating and deleting is not implemented yet.

**Comment by PVince81** [comment]
While splitting the "old public webdav" and "new public webdav" I came accross a test tagged with `@issue-36055`.
That test had the test steps commented out for the "new public webdav".

It turns out that this bit still doesn't work. I've now marked it as `@skipOnOcV10` and reopened the ticket, see  https://github.com/owncloud/core/issues/36055#issuecomment-648895940

@individual-it FYI

**Comment by PVince81** [comment]
I ran the unskipped tests marked with issue 49 against the matching Reva PR that solves those: https://github.com/cs3org/reva/pull/877
```
TEST_SERVER_URL=http://localhost:9140 \
TEST_EXTERNAL_USER_BACKENDS=true \
TEST_OCIS=true \
OCIS_REVA_DATA_ROOT=/var/tmp/reva/ \
SKELETON_DIR=$HOME/work/workspace/owncloud/apps-external/testing/data/apiSkeleton \
BEHAT_FILTER_TAGS='~@skipOnOcV10&&~@skipOnLDAP&&@TestAlsoOnExternalUserBackend&&~@local_storage&&@issue-ocis-reva-49&&~@skipOnOcis'
```

and they all pass:
```
21 scenarios (21 passed)
145 steps (145 passed)
```

please note that some are still skipped due to other issues, see OP for a listing of related issues to address separately.

**Comment by PVince81** [comment]
we might want to wait for the Reva PRs to be merged and updated in ocis-reva just to be on the safe side

File: tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature [hunk 1]
```diff
@@ -27,17 +27,44 @@ Feature: create a public link share
...
+      | file_target            | /randomfile.txt |
+      | path                   | /randomfile.txt |
+      | permissions            | read            |
+      | share_type             | public_link     |
+      | displayname_file_owner | %displayname%   |
+      | displayname_owner      | %displayname%   |
+      | uid_file_owner         | Alice           |
+      | uid_owner              | Alice           |
+      | name                   |                 |
+    When the public downloads the last public shared file with range "bytes=0-9" using the old public WebDAV API
```
Comment (individual-it) [inline tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature:57]:
```suggestion
    When the public downloads the last public shared file with range "bytes=0-9" using the new public WebDAV API
```

File: tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature [hunk 2]
```diff
@@ -27,17 +27,44 @@ Feature: create a public link share
...
     Then the downloaded content should be "Random dat"
     And the public upload to the last publicly shared file using the old public WebDAV API should fail with HTTP status code "403"
```
Comment (individual-it) [inline tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature:59]:
```suggestion
    And the public upload to the last publicly shared file using the new public WebDAV API should fail with HTTP status code "403"
```

File: tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature [hunk 3]
```diff
@@ -27,17 +27,44 @@ Feature: create a public link share
...
-    And the public upload to the last publicly shared file using the new public WebDAV API should fail with HTTP status code "403"
+    Then the downloaded content should be "Random dat"
```
Comment (individual-it) [inline tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature:60]: remove this line, that check is done above and fail here because there is no download before this step

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by PVince81** [comment]
@individual-it thanks for spotting. I already rechecked those bits over and over again for discrepancies but still managed to miss it. I've fixed that location and squashed.

**Review by individual-it — APPROVED** [review]

**Comment by PVince81** [comment]
API tests passed in https://cloud.drone.io/owncloud/ocis-reva/521, merging
